cpp/src/io/parquet/page_enc.cu

```diff
@@ -606,36 +804,75 @@ inline __device__ void PackLiterals(
   uint8_t *dst, uint32_t v, uint32_t count, uint32_t w, uint32_t t)
 {
   if (t <= (count | 0x1f)) {
```

Comment:
Now that there is a _syncthreads(), this `if (t <= (count | 0x1f))` needs to go away (deadlock between syncthreads amd early return statements)

Reply:
But there's another `return` for this very purpose. https://github.com/rapidsai/cudf/blob/85cd56dfb3449140f18c7cef3a3be01ac976fd14/cpp/src/io/parquet/page_enc.cu#L844-L846

Reply:
Oh I see. I think it'd be simpler to just removing both and just move the generic atomic-based path in a else {} block (that would also allow getting rid of all the return statements).